src/map/MapSidebar.js

```diff
@@ -105,31 +105,8 @@ function MapSidebarMain (props) {
   const [tentativeGrowth, setTentativeGrowth] = useState([]);
   // gets the current number of Reputables
   useEffect(() => {
-    async function getReputableCount () {
-      const CommunityCeremony = api.registry.getOrUnknown('CommunityCeremony');
-
-      const [reputationLifetime, currentCeremonyIndex] = await Promise.all([
-        api.query.encointerCeremonies.reputationLifetime(),
-        api.query.encointerScheduler.currentCeremonyIndex()
-      ]);
-
-      const promises = [];
-      const lowerIndex = Math.max(0, currentCeremonyIndex - reputationLifetime);
-
-      for (let cIndex = lowerIndex; cIndex <= currentCeremonyIndex; cIndex++) {
-        const communityCeremony = new CommunityCeremony(api.registry, [cid, cIndex]);
-        promises.push(api.query.encointerCeremonies.participantReputation.keys(communityCeremony));
-      }
-
-      const arrayOfReputablesArray = await Promise.all(promises);
-
-      // reduce the array of arrays to a single set.
-      const allReputablesSet = new Set(arrayOfReputablesArray.reduce((all, nextArray) => [...all, ...nextArray]));
-
-      setAllReputableNumber(allReputablesSet.size);
-    }
-    getReputableCount();
-  }, [api, cid]);
+    getAllReputables(api, cid, debug).then((reputables) => setAllReputableNumber([reputables.length]));
+  }, [api, cid, debug]);
```

Comment:
@Jay-Forrer 

This is how I want to have the code structure. The use `effectEffect` calls a single function or maybe two or three functions that are stateless aka everything they need is supplied as a function arg. Then the return value of those functions can be use for the `setEffects` function